From d3bfc23408ac6f0b32fb7c369d3665598d2bb3d1 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Fri, 28 Jun 2024 14:08:40 +0800
Subject: [PATCH 53/67] BACKPORT: DEEPIN: net: ethernet: fix phytmac on 6.9

Declare multiple functions as static functions to suppress warnings about
missing prototypes.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>

Link: https://github.com/deepin-community/kernel-rolling/commit/3672205e780ee88d27b7885e27f450cd332f842c
[Kexy: Resolved minor conflicts in
 drivers/net/ethernet/phytium/phytmac_main.c, and
 drivers/net/ethernet/phytium/phytmac_v2.c]
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/net/ethernet/phytium/phytmac_main.c     | 8 ++++----
 drivers/net/ethernet/phytium/phytmac_platform.c | 1 -
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_main.c b/drivers/net/ethernet/phytium/phytmac_main.c
index 9ed5ec09523b..6ef55093d6d4 100644
--- a/drivers/net/ethernet/phytium/phytmac_main.c
+++ b/drivers/net/ethernet/phytium/phytmac_main.c
@@ -1848,7 +1848,7 @@ static int phytmac_phylink_connect(struct phytmac *pdata)
 	return 0;
 }

-int phytmac_pcs_config(struct phylink_pcs *pcs, unsigned int mode,
+static int phytmac_pcs_config(struct phylink_pcs *pcs, unsigned int mode,
 		       phy_interface_t interface,
 		       const unsigned long *advertising,
 		       bool permit_pause_to_mac)
@@ -1856,7 +1856,7 @@ int phytmac_pcs_config(struct phylink_pcs *pcs, unsigned int mode,
 	return 0;
 }

-void phytmac_pcs_link_up(struct phylink_pcs *pcs, unsigned int mode,
+static void phytmac_pcs_link_up(struct phylink_pcs *pcs, unsigned int mode,
 			 phy_interface_t interface, int speed, int duplex)
 {
 	struct phytmac *pdata = container_of(pcs, struct phytmac, phylink_pcs);
@@ -2006,7 +2006,7 @@ static void phytmac_mac_link_up(struct phylink_config *config,
 	netif_tx_wake_all_queues(ndev);
 }

-int phytmac_mdio_register(struct phytmac *pdata)
+static int phytmac_mdio_register(struct phytmac *pdata)
 {
 	struct phytmac_hw_if *hw_if = pdata->hw_if;
 	int ret;
@@ -2648,7 +2648,7 @@ static int phytmac_init(struct phytmac *pdata)
 	return 0;
 }

-void phytmac_default_config(struct phytmac *pdata)
+static void phytmac_default_config(struct phytmac *pdata)
 {
 	struct net_device *ndev = pdata->ndev;

diff --git a/drivers/net/ethernet/phytium/phytmac_platform.c b/drivers/net/ethernet/phytium/phytmac_platform.c
index ce6271ac52cd..cf628a451728 100644
--- a/drivers/net/ethernet/phytium/phytmac_platform.c
+++ b/drivers/net/ethernet/phytium/phytmac_platform.c
@@ -168,7 +168,6 @@ static int phytmac_get_phy_mode(struct platform_device *pdev)
 static int phytmac_plat_probe(struct platform_device *pdev)
 {
 	const struct phytmac_config *phytmac_config = &phytium_1p0_config;
-	struct device_node *np = pdev->dev.of_node;
 	struct resource *regs;
 	struct phytmac *pdata;
 	int ret, i;
--
2.50.1
